File: uploads/job_1777126575944/student3/Calculator.js
class Calculator {

    // reused concept: addition
    add(x, y) {
        return x + y;
    }

    // modified logic: safe division with error handling
    divide(a, b) {
        if (b === 0) {
            console.log("Cannot divide by zero");
            return null;
        }
        return a / b;
    }

    // changed implementation: recursive power instead of loop
    power(base, exp) {
        if (exp === 0) return 1;
        return base * this.power(base, exp - 1);
    }

}

// main execution
const calc = new Calculator();

console.log("Power result:", calc.power(2, 3));
console.log("Addition:", calc.add(5, 7));
console.log("Division:", calc.divide(10, 2));

File: uploads/job_1777126575944/student2/Calculator.java
public class Calculator {

    // reused logic (similarity part)
    public int add(int x, int y) {
        return x + y;
    }

    // changed logic (reduces similarity)
    public int divide(int a, int b) {
        if (b == 0) return 0;
        return a / b;
    }

    public int power(int base, int exp) {
        int result = 1;
        for (int i = 0; i < exp; i++) {
            result *= base;
        }
        return result;
    }

    public static void main(String[] args) {
        Calculator calc = new Calculator();
        System.out.println(calc.power(2, 3));
    }
}
